tsdb/record/record_test.go

```diff
@@ -385,3 +468,140 @@ func TestRecord_MetadataDecodeUnknownExtraFields(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, expectedMetadata, decMetadata)
 }
+
+type recordsMaker struct {
+	name string
+	init func(int, int, int)
+}
+
+// BenchmarkWAL_HistogramLog measures efficiency of encoding classic
+// histograms and native historgrams with custom buckets (NHCB).
+func BenchmarkWAL_HistogramEncoding(b *testing.B) {
+	// Cache for the refs.
+	var series []RefSeries
+	var samples []RefSample
+	var nhcbs []RefHistogramSample
+
+	resetCache := func() {
+		series = nil
+		samples = nil
+		nhcbs = nil
+	}
+
+	initClassicRefs := func(labelCount, histograms, buckets int) {
```

Comment:
Can we just do a classic function following some  `type refsCreateFn func(labelCount, histograms, buckets int) ([]RefSeries, []RefSample, []RefHistogramSample)`? Let's avoid closure when it's not needed, WDYT?